kafka-cron/README.md

```diff
@@ -0,0 +1,162 @@
+# Distributed Cron Based on Kafka
+
+In this project we're going to build a simple distributed `cron` system, based on the Apache Kafka distributed queue system.
+
+> **Note:** This project requires us to have Docker with [Compose](https://docs.docker.com/compose/) installed. Compose is a tool for defining and running multi-container Docker applications. With Compose, we use a YAML file to configure the application’s services. Then, with a single command, we create and start all the services from the configuration.
+
+Learning Objectives
+ - How can we use a distributed queue in software architecture?
+ - How can we deal with errors in a system based on distributed queues?
+ - How can we instrument a complex application with metrics? How should we design alerting?
+
+Timebox: 3 days
```

Comment:
I would probably call this a 5 day project - there are a lot of moving pieces to get to grips with, particularly in the monitoring section. Thinking back to the first time I tried to configure prometheus (and get to grips with promql!), alertmanager, and grafana - there's a lot there before you can even start thinking about what's good/bad ideas.

I'd possibly consider adding a couple of sample promql queries that could be useful examples (maybe a "p90/p100 time for consumer to execute job" and a "p99 lateness of dequeuing") as inspiration - I can believe if you've never written promql before, it could take a long and disheartening time to write your first meaningful query, and that that time and frustration could stop people from getting to the more interesting consideration of "what's a good metric?"

Reply:
did one example, pointed at a page of other examples.

Reply:
will make myself available to help as well.